File: Implementation/Mathematics/NumberTheory/java/PrimalityTest.java
import java.util.Scanner;

public class PrimalityTest {
    public static boolean isPrime(int n) {
        if (n <= 1)
            return false;
        if (n <= 3)
            return true;

        if (n % 2 == 0 || n % 3 == 0)
            return false;

        for (int i = 5; i * i <= n; i += 6) {
            if (n % i == 0 || n % (i + 2) == 0)
                return false;
        }

        return true;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n = sc.nextInt();
        if (isPrime(n))
            System.out.println("Yes");
        else
            System.out.println("No");
    }
}

File: Implementation/Mathematics/NumberTheory/python/PrimalityTest.py
def isPrime(n):
    if n <= 1:
        return False
    if n <= 2:
        return True

    if n % 2 == 0 or n % 3 == 0:
        return False

    counter = 5
    while (counter * counter <= n):
        if n % counter == 0 or n % counter + 2:
            return False

        counter += 6

    return True

n = int(input())

if (isPrime(n)):
    print("Yes")
else:
    print("No")
